dart-parser/src/dart/maybe_required.rs

```diff
@@ -0,0 +1,58 @@
+use core::ops::Deref;
+use std::fmt::Debug;
+
+pub struct MaybeRequired<T> {
+    pub is_required: bool,
+    value: T,
+}
+
+impl<T> MaybeRequired<T> {
+    pub fn new(is_required: bool, value: T) -> Self {
+        Self { is_required, value }
+    }
+
+    pub fn required(value: T) -> Self {
+        Self {
+            is_required: true,
+            value,
+        }
+    }
+
+    pub fn optional(value: T) -> Self {
+        Self {
+            is_required: false,
+            value,
+        }
+    }
+}
+
+impl<T> Deref for MaybeRequired<T> {
```

Comment:
This is probably [something to get rid of](https://rust-unofficial.github.io/patterns/anti_patterns/deref.html).